rele/client.py

```diff
@@ -161,6 +161,10 @@ def publish(self, topic, data, blocking=False, timeout=None, **attrs):
         if not blocking:
             return future
 
-        future.result(timeout=timeout or self._timeout)
+        try:
+            future.result(timeout=timeout or self._timeout)
+        except TimeoutError as e:
```

Comment:
Does it raise a builtin Timeout or its own custom Timeout that we need to catch? IIRC, it is the latter case.

Reply:
You're right. It comes from concurrent.futures:
https://github.com/googleapis/python-pubsub/blob/e204b86e34346b6d456771ef4d4cfe5e15e53238/google/cloud/pubsub_v1/exceptions.py#L17